core/autocrypt/cmdline.py

```diff
@@ -205,16 +205,13 @@ def process_incoming(ctx):
     """parse autocrypt headers from stdin mail. """
     account = get_account(ctx)
     msg = mime.parse_message_from_file(sys.stdin)
-    peerinfo = account.process_incoming(msg)
-    if peerinfo is not None:
-        click.echo("processed mail for identity '{}', found: {}".format(
-                   peerinfo.identity.config.name, peerinfo))
+    r = account.process_incoming(msg)
+    if r.peerstate.autocrypt_timestamp == r.peerstate.last_seen:
```

Comment:
I think instead of returning and accessing the peerstate in process_incoming we should preprocess the mail, extract the header and create an intermediate object. We can then hand this object to the peerstate to process it and return it. It would represent the new change we just send in for processing rather than the aggregated result.